pythonx/jedi_vim.py

```diff
@@ -158,42 +158,84 @@ def wrapper(*args, **kwargs):
     return func_receiver
 
 
+# XXX: might use lru cache, given that you can configure it per buffer?!
 current_environment = (None, None)
 
 
+def set_environment(executable=None, bufnr=None):
+    global current_environment
+
+    # print(repr(executable), repr(bufnr))
+
+    if not executable:  # "unset"
+        executable = 'auto'
+        environment = None
+    else:
+        try:
+            # XXX: create_environment is not deterministic
+            #      (https://github.com/davidhalter/jedi/issues/1155)
+            # jedi.create_environment(executable, safe=False)
+            environment = jedi.api.environment.Environment(executable)
+        except jedi.InvalidPythonEnvironment as exc:
+            echo_highlight('executable=%s is not supported: %s.' % (
+                executable, str(exc)))
+            return
+    bufnr = int(bufnr)
+    if bufnr:
+        vim_command("call setbufvar(%d, 'jedi_use_environment', %r)" % (
+                    bufnr, executable))
+    else:
+        vim_command("let g:jedi#use_environment = %r" % executable)
+    current_environment = (executable, environment)
+
+
 def get_environment(use_cache=True):
```

Comment:
Add `scope` here then, too.  But it would default to preferring bufnr over window over tab over globals.